Dynatrace/dynatrace-operator
# Auto-detects platform for helm install

# Description

Using the available API versions on the target cluster we can determine the platform so the user don't have to set it manually.

The "mapping" is the following:

security.openshift.io/v1 ==> openshift
auto.gke.io/v1 ==> gke-autopilot
fallback ==> kubernetes

The .Values.platform is still available to override if something goes wrong.
Furthermore, this "auto-detect" feature does not work for helm template, which we use for development.
(this is perfectly fine, and makes all the sense)

## How can this be tested?
Run helm install dynatrace-operator config/helm/chart/default ... on all the platforms to see if everything is applied correctly. (make sure to set an image that actually exists)


## Checklist
